library/src/test/java/com/novoda/downloadmanager/CallbackThrottleByProgressIncreaseTest.java

```diff
@@ -0,0 +1,67 @@
+package com.novoda.downloadmanager;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.*;
+
+public class CallbackThrottleByProgressIncreaseTest {
+
+    private static final int DOWNLOAD_PERCENTAGE = 75;
+
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    private final DownloadBatchCallback downloadBatchCallback = mock(DownloadBatchCallback.class);
+    private final DownloadBatchStatus downloadBatchStatus = mock(DownloadBatchStatus.class);
+
+    private CallbackThrottleByProgressIncrease callbackThrottleByProgressIncrease;
+
+    @Before
+    public void setUp() {
+        callbackThrottleByProgressIncrease = new CallbackThrottleByProgressIncrease();
+    }
+
+    @Test
+    public void throwsException_whenStoppingUpdates_withoutSettingCallback() {
+        thrown.expect(NullPointerException.class);
+
+        callbackThrottleByProgressIncrease.stopUpdates();
+    }
+
+    @Test
+    public void doesNothing_whenUpdating_withoutSettingCallback() {
```

Comment:
the underscores seem a little arbitrary... I guess because the names are still quite long. I get that they're not very easy to name tho.

Reply:
They aren't arbitrary 😄  

`expectedBehavior_whenStateUnderTest_furtherClauses` obviously trying to avoid the `FurtherClausesPart` 😄

Reply:
how c/would you name this one?

for me the underscores seem legit, separating each chunk of information. would you have preferred no underscores?

Reply:
yeah, would prefer getting the names short enough to not have underscores. To me they're a smell the name is still too long